R/dist_spec.R

```diff
@@ -1099,3 +1101,84 @@ convert_to_natural <- function(params, distribution) {
   }
   return(params)
 }
+
+##' Perform checks for `<dist_spec>` `get_...` functions
+##'
+##' @param x A ``<dist_spec>``.
+##' @param id Integer; the id of the distribution to get parameters of (if x is
+##' a composite distribution). If `x` is a single distribution this is ignored
+##' and can be left as `NULL`.
+##' @return The id to use.
+##' @keywords internal
+##' @author Sebastian Funk
+check_get_dist_spec <- function(x, id) {
+  if (!is(x, "dist_spec")) {
+    stop("Can only get parameters of a <dist_spec>.")
+  }
+  if (!is.null(id) && id > length(x)) {
+    stop(
+      "`id` can't be greater than the number of distributions (", length(x),
+      ")."
+    )
+  }
+  if (length(x) > 1) {
+    if (is.null(id)) {
+      stop("`id` must be specified when `x` is a composite distribution.")
+    }
+  } else {
+    id <- 1
+  }
+  return(id)
+}
```

Comment:
This looks like it combines a "checking" with another functionality that makes an id? I suggest aligning this with how our other "checks" work, i.e., return an error if something is wrong, or invisibly return `x`. 

Maybe split this into a "check_id()" and "generate_id()". The latter will call the check function first and then return `id`.
 
Also, should this live in the `checks.R` script?

Reply:
Good point - I've renamed this to `get_dist_spec_id()` in [2b0565c](https://github.com/epiforecasts/EpiNow2/pull/646/commits/2b0565c6f9d15d59a6e41e1948874e2940350d5f). Could still ship out the first check but maybe it's fine in there?